src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
```

Comment:
camelCase should be used. Correct name would be definedCapacity. You should shorten the words that might have more than one meaning, like defined or definitive in this case. please check google java style guide for more information about naming, for example here:
https://mate-academy.github.io/style-guides/java/java.html#s5.3-camel-case